lib/Munin/Node/Configure/Plugin.pm

```diff
@@ -75,6 +75,16 @@ sub suggestion_string
         # Report why it's not being used
         $msg = " [$self->{defaultreason}]";
     }
+    elsif (! $self->{capabilities}->{autoconf}) {
```

Comment:
I think, `autoconf` is not a strict requirement for `suggest`, or?
Thus I think, this message would be misleading.

Reply:
I added `&& ! $self->{capabilities}->{suggest}`
The idea is that when no autoconf is supported the default `no` value does not get overridden by a plugin run

Reply:
OK. Sorry - my first attempt of understanding that change failed :(
Now I see, that the function is called `suggestion_string`, but indeed it is used for `autoconf` (and not for `suggest`).

Thus you should ignore my previous comment.

Since I do not really grasp the flow of the `suggestion_string` function: could you please explain, which problem your change solves?
I am not a user of that plugin module and I have problems understanding what the fields `default` and `defaultreason` are supposed to do and why we should not check for `{capabilities}->{autoconf}` straight at the beginning of the function (or even before calling it).

Reply:
`suggestion_string` is called from `munin-node-configure` when using the `--suggest` flag.
`->default` is the autoconf answer ("yes"/"no") and `->defaultreason` the autoconf description. (e.g. `echo "no (command foo not found)"` results in `default` equals "no" and `defaultreason` equals "command foo not found")

E.g.:
```
$ ./dev_scripts/install node
$ ./dev_scripts/run munin-node-configure --suggest --families=manual
Found: ...munin/sandbox/bin/munin-node-configure
Command line:  munin-node-configure --suggest --families=manual
Plugin                     | Used | Suggestions                            
------                     | ---- | -----------                            
apt                        | no   | yes                                    
apt_all                    | no   | yes                                    
bind9_rndc                 | no   | no
...
```

With my change, the `no` is further explained:

```
./dev_scripts/run munin-node-configure --suggest --families=manual
# [dev/run] Found: .../munin/sandbox/bin/munin-node-configure
# [dev/run] Command line:  munin-node-configure --suggest --families=manual
Plugin                     | Used | Suggestions                            
------                     | ---- | -----------                            
apt                        | no   | yes                                    
apt_all                    | no   | yes                                    
bind9_rndc                 | no   | no [[[ plugin has neither autoconf not suggest support ]]]
...
```